src/main/java/com/spotify/dns/ServiceResolvingChangeNotifier.java

```diff
@@ -116,16 +119,25 @@ public void run() {
       current = builder.build();
     } catch (Exception e) {
       log.error(e.getMessage(), e);
+      fireEmpty();
       return;
     }
 
-    if (!current.equals(records)) {
+    if (waitingForFirstEvent || !current.equals(records)) {
+      waitingForFirstEvent = false;
       final ChangeNotification<T> changeNotification =
           newChangeNotification(current, records);
       records = current;
 
       fireRecordsUpdated(changeNotification);
     }
   }
+
+  private void fireEmpty() {
```

Comment:
This method name doesn't feel great - `checkFireFirst`, `maybeFireInitialUpdate` or something? It sounds like it will always fire, but in fact, it will only fire if it hasn't been fired ever. Also, 'empty' feels like it could become wrong, given that it's reading using the `current()` method.